linbox/matrix/polynomial-matrix.h

```diff
@@ -293,142 +325,159 @@ namespace LinBox{
-		const Field*       _fld;
+    private:
+        Data     _rep;
+        size_t   _row;
+		size_t   _col;
+		size_t  _size;
 	};
 
-	// Class for Polynomial Matrix stored as a Polynomial of Matrices
-	template<class _Field>
-	class PolynomialMatrix<PMType::matfirst,PMStorage::plain,_Field> {
-	public:
-		typedef _Field Field;
-		typedef typename Field::Element   Element;
-		typedef BlasMatrix<Field>         Matrix;
-		typedef std::vector<Element>       Polynomial;
-		typedef PolynomialMatrix<PMType::matfirst,PMStorage::plain,_Field>  Self_t;
-		typedef PolynomialMatrix<PMType::polfirst,PMStorage::plain,_Field> Other_t;		
-			       
-		typedef Matrix value_type;
-
-		PolynomialMatrix() {}
-
-		PolynomialMatrix(const Self_t&) = delete;
-		
-		PolynomialMatrix(const Field& f, size_t r, size_t c, size_t s) :
-			_rep(s,Matrix(f)), _row(r), _col(c), _size(s), _fld(&f) {
-			//_row(r), _col(c), _size(s), _fld(&f) {			
-			for(size_t i=0;i<s;i++)
-				_rep[i].init(r,c);
-			//integer p;
-			//std::cout<<"(ALLOC) matfirst at "<<this<<" : "<<r<<"x"<<c<<" - size= "<<s<<" ==> "<<MB(realmeminfo())<<" Mo"<<std::endl;
-			ADD_MEM(realmeminfo());
-		}
+    // Class for Polynomial Matrix stored as a Polynomial of Matrices
+    template<class _Field>
+	class PolynomialMatrix<_Field, PMType::matfirst> {
+    public:
+        typedef _Field                         Field;
+		typedef typename Field::Element      Element;
+		typedef std::vector<Element,AlignedAllocator<Element, Alignment::DEFAULT>> VECT;
+        typedef BlasVector<Field>                             Data;        
 
-		~PolynomialMatrix(){
-			DEL_MEM(realmeminfo());
-			//std::cout<<"(FREE) matfirst at "<<this<<" : "<<_row<<"x"<<_col<<" - size= "<<_size<<" ==> "<<MB(realmeminfo())<<" Mo"<<std::endl;
-			//integer p;
-			//std::cout<<"PMatrix Desallocating : "<<_row*_col*_size*length(_fld->characteristic(p))/1000000.<<"Mo"<<std::endl;
-		}
+        typedef BlasSubmatrix<BlasMatrix<Field>>                    Matrix;
+        typedef BlasSubmatrix<const BlasMatrix<Field>>         constMatrix;
+        typedef BlasSubvector<Data>                             Polynomial;
+        typedef const BlasSubvector<const Data>            constPolynomial;        
 
-		void clear(){
-			_rep.resize(0,Matrix(*_fld));
-		}
+		typedef PolynomialMatrix<_Field, PMType::matfirst>  Self_t;
+        typedef SubPolynomialMatrix<Self_t>                   view;
+        typedef SubPolynomialMatrix<const Self_t>       const_view;
+        
+		// construct a polynomial matrix in f[x]^(m x n) of degree (s-1)
+        PolynomialMatrix(const Field& f, size_t r, size_t c, size_t s)
+            : _rep(f,r*c*s), _row(r), _col(c), _size(s) {}
+
+
+        /*******************************
+         * Data access functionnalities*
+         ******************************/
+        
+        // retrieve the polynomial at entry (i,j) in the matrix
+		Polynomial       operator()(size_t i, size_t j)      {return operator()(i*_col+j);}
```

Comment:
same remark wrt getEntry and refEntry here

Reply:
see discussion above https://github.com/linbox-team/linbox/pull/285#discussion_r766559416